src/posthog-surveys.ts

```diff
@@ -76,10 +76,20 @@ export class PostHogSurveys {
     }
 
     onRemoteConfig(response: RemoteConfig) {
-        this._decideServerResponse = !!response['surveys']
-        logger.info(`decideServerResponse set to ${this._decideServerResponse}`)
+        // only load surveys if they are enabled and there are surveys to load
+        const surveys = response['surveys']
+        if (isArray(surveys)) {
+            this._hasSurveys = surveys.length > 0
+        } else if (isBoolean(surveys)) {
+            this._hasSurveys = surveys
+        } else {
+            logger.warn('Decide not loaded yet. Not loading surveys.')
+        }
 
-        this.loadIfEnabled()
+        if (this._hasSurveys) {
+            logger.info(`decide response received, hasSurveys: ${this._hasSurveys}`)
+            this.loadIfEnabled()
+        }
```

Comment:
feels like we can simplify this a bit. what do you think?

```ts
    onRemoteConfig(response: RemoteConfig) {
        // only load surveys if they are enabled and there are surveys to load
        const surveys = response['surveys']
        if (isUndefined(surveys)) {
            return logger.warn('Decide not loaded yet. Not loading surveys.')
        }
        const isArrayResponse = isArray(surveys)
        this._hasSurveys = isArrayResponse ? surveys.length > 0 : surveys
        if (this._hasSurveys) {
            logger.info(`decide response received, hasSurveys: ${this._hasSurveys}`)
            this.loadIfEnabled()
        }
    }
```

Reply:
actually i'd move the decide response received log to be outside the `if` so we always log the response

Reply:
yep, thx